beacon-chain/types/block.go

```diff
@@ -58,6 +58,10 @@ func (b *Block) Hash() ([32]byte, error) {
 // ParentHash corresponding to parent beacon block.
 func (b *Block) ParentHash() [32]byte {
 	var h [32]byte
+	if len(b.data.ParentHash) != 32 {
```

Comment:
Is this a sanity check? Why do we want to return an empty byte array when `ParentHash` isn't 32 bytes long?

Reply:
I agree that returning an empty array is not a good signal, however it's something that can be checked by callers. Without this the `copy` can panic, which I thought is worse. Should I return an err instead?

Reply:
Maybe this validation could go into `NewBlockWithData`, which does return an err, and assume in `ParentHash` that the data is indeed 32 bytes.

Reply:
Oh, I see. You bring up a good point. The reason `b.data.ParentHash` isn't a byte array to begin with is that protobufs don't support fixed length arrays. This should always be 32 bytes, so yea, can you please check that `ParentHash` is 32 bytes inside `NewBlockWithData`?